### DISCUSSION TITLE
LCP spent too long time

### DISCUSSION THREAD

**Discussion by nuanyang233** [vercel/next.js#20464] [Show and tell]
![image](https://user-images.githubusercontent.com/28348152/103123532-a1ab7b80-46bf-11eb-893d-2c9f6f86f6a0.png)

I was recently build my site with nextjs and when I deployed it in vercel and tested it with pageSpeed, I found that the LCP was very high.

So I tested nextjs.org with pageSpeed and found that the LCP was just as high in the lab.

Can anyone tell me what could be the cause of this result? Is it a problem with nextjs? Or is it a Google PageSpeed problem?

**Comment by awareness481** [comment]
Keep in mind that a single test isn't enough to determine the exact score, as the score can vary from to test to test. You should also make sure to run the tests in Incognito mode, so that no extensions hurt your score. Extensions like React Dev Tools, Web Vitals, etc will hurt your score because Lighthouse thinks the script files come from your website.

Here's the results from the Lighthouse audit I ran just now:

![image](https://user-images.githubusercontent.com/12380586/103125125-d523e100-4692-11eb-8ec6-6bed5d924dec.png)

Edit: The `Image elements do not have explicit width and height` is a bug of Lighthouse when using the next/image component. See: https://github.com/GoogleChrome/lighthouse/issues/11631

Edit 2: The above were for desktop, here are for mobile.

![image](https://user-images.githubusercontent.com/12380586/103125526-051fb400-4694-11eb-8b51-77fa6e66057e.png)

Not perfect but I'm sure there's room for improvement. You can certainly get perfect score using Next.js, I'd look into the warnings Lighthouse gives you and try to identify the issues.

**Reply by svassaux** [reply]
that's just not good. if the content is served from the server, we can't be happy with an ok score, we should have a 100 score and a good rehydration or script loading

**Comment by svassaux** [comment]
I have the same issue as you and I'm trying hard to find solutions

There is a huge problem with rehydration and the way scripts are loaded by nextjs
It's known but not acknowledged by everybody: https://twitter.com/devongovett/status/1222953658049974272

Here are my issues with some response on how to improve your perf:
https://github.com/vercel/next.js/discussions/20454
https://github.com/vercel/next.js/discussions/20700

**Comment by Confuze** [comment]
It looks like lighthouse's simulated throttling may be exaggerating the LCP.

Here's my result with simulated throttling:
![image](https://github.com/user-attachments/assets/a473771c-535f-48d2-a078-f255a67bd45d)

And here it is with devtools throttling:
![image](https://github.com/user-attachments/assets/d98a6c67-e0a5-4adc-9658-7348f779c2e0)

Both test were done on the mobile option so the difference is pretty big.
